addons/website_sale/models/product_pricelist.py

```diff
@@ -122,10 +106,18 @@ def _get_website_pricelists_domain(self, website):
         ''' Check above `_is_available_on_website` for explanation.
         Change in this method should be reflected in `_is_available_on_website`.
         '''
+        if website.company_id:
+            default_pricelist = website.company_id._get_default_pricelist_vals()
         return [
             ('active', '=', True),
             ('company_id', 'in', [False, website.company_id.id]),
-            '|', ('website_id', '=', website.id),
-            '&', ('website_id', '=', False),
-            '|', ('selectable', '=', True), ('code', '!=', False),
+            '|', '|', ('website_ids', 'any', [('id', '=', website.id)]),
+            '&', ('website_ids', '=', False),
+            ('code', '!=', False),
+            # If default pricelist
+            ('name', '=', default_pricelist['name']),
```

Comment:
unreliable as it'll change according to the current user/website lang, why do you need the default pricelist ?